### ISSUE TITLE
Unit test failure with TF_ENABLE_ONEDNN_OPTS=1

### ISSUE THREAD

**Issue by elfringham** [tensorflow/tensorflow#59773]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Have you reproduced the bug with TF nightly?

Yes

### Source

source

### Tensorflow Version

git HEAD

### Custom Code

No

### OS Platform and Distribution

Ubuntu 20.04

### Mobile device

n/a

### Python version

3.9.13

### Bazel version

5.3.0

### GCC/Compiler version

10.3.0

### CUDA/cuDNN version

n/a

### GPU model and memory

n/a

### Current Behaviour?

```shell
//tensorflow/python/kernel_tests/nn_ops:pooling_ops_test_cpu             FAILED in 3 out of 12 in 20.7s
```

### Standalone code to reproduce the issue

```shell
bazel test --test_timeout=300,500,-1,-1 --flaky_test_attempts=3 --test_output=all --cache_test_results=no --noremote_accept_cached --test_env=TF_ENABLE_ONEDNN_OPTS=1 --build_tag_filters=-no_oss,-oss_serial,-gpu,-tpu,-benchmark-test,-v1only,-requires-gpu --test_tag_filters=-no_oss,-oss_serial,-gpu,-tpu,-benchmark-test,-v1only,-requires-gpu --verbose_failures --build_tests_only --jobs=16 -- //tensorflow/python/kernel_tests/nn_ops:pooling_ops_test
```

### Relevant log output

```shell
FAIL: testAvgPoolGradOutputMemoryOutOfBounds (__main__.PoolingTest)
PoolingTest.testAvgPoolGradOutputMemoryOutOfBounds
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/builder/.cache/bazel/_bazel_builder/945690c41481150b9aa58576637dd867/execroot/org_tensorflow/bazel-out/aarch64-opt/bin/tensorflow/python/kernel_tests/nn_ops/pooling_ops_test_cpu.runfiles/org_tensorflow/tensorflow/python/kernel_tests/nn_ops/pooling_ops_test.py", line 2348, in testAvgPoolGradOutputMemoryOutOfBounds
    self.evaluate(
AssertionError: InvalidArgumentError not raised
```
</details>

**Comment by elfringham** [comment]
@TensorFlow-MKL

**Comment by elfringham** [comment]
@cfRod @nSircombe

**Comment by elfringham** [comment]
This does not happen with protobuf==3.20.3 installed. However the default install currently will be protobuf==4.22.0

**Comment by agramesh1** [comment]
@elfringham thanks for reporting, we will take a look at it.

**Comment by nazneenn** [comment]
Hi @elfringham, we can reproduce this issue internally with commit id - 62dcf7e02c0dae336bfb54aa83a03ddd22ccc9af

Thank you for your report, we will continue to investigate it and update you if any workaround 
In the meanwhile could you also verify if the git commit id is the same since the master branch is updated everyday?

**Comment by SuryanarayanaY** [comment]
@agramesh1 , Thanks for quick response. Please keep us updated. Thankyou!

**Comment by elfringham** [comment]
@nazneenn You can see that the ARM_CI run for that commit has the failing test in it.
https://github.com/tensorflow/tensorflow/actions/runs/4246640900/jobs/7383702056
However please be aware that since then this test is now being excluded from ARM_CI runs. If you put a PR in to fix the issue then please also remove the line https://github.com/tensorflow/tensorflow/blob/b5154bce3272d6ba430a881c92e017eb15408cad/tensorflow/tools/ci_build/build_scripts/ARM_SKIP_TESTS.sh#L30 so that the test will be run again.

**Comment by nazneenn** [comment]
@elfringham ,[ PR has been created which fixes this issue  #59837 ](https://github.com/tensorflow/tensorflow/pull/59837)

**Comment by elfringham** [comment]
Fixed by merge of #59837

**Comment by SuryanarayanaY** [comment]
I cross checked and found now the tests are success.
```

----------------------------------------------------------------------
Ran 97 tests in 8.219s

OK (skipped=61)
/bin/bash: /home/suryanarayanay/miniconda3/lib/libtinfo.so.6: no version information available (required by /bin/bash)
================================================================================
Target //tensorflow/python/kernel_tests/nn_ops:pooling_ops_test_cpu up-to-date:
  bazel-bin/tensorflow/python/kernel_tests/nn_ops/pooling_ops_test_cpu
INFO: Elapsed time: 11.607s, Critical Path: 10.52s
INFO: 11 processes: 1 internal, 10 local.
INFO: Build completed successfully, 11 total actions
//tensorflow/python/kernel_tests/nn_ops:pooling_ops_test_cpu             PASSED in 10.5s
  Stats over 10 runs: max = 10.5s, min = 2.3s, avg = 4.6s, dev = 3.0s

Executed 1 out of 1 test: 1 test passes.
There were tests whose specified size is too big. Use the --test_verbose_timeout_warnings command li
INFO: Build completed successfully, 11 total actions

```

Full log attached below.
[59773_logs.txt](https://github.com/tensorflow/tensorflow/files/11300417/59773_logs.txt)
